sdk/monitor/monitor-ingestion/src/logsIngestionClient.ts

```diff
@@ -0,0 +1,146 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+import { TokenCredential } from "@azure/core-auth";
+import { CommonClientOptions } from "@azure/core-client";
+import { SDK_VERSION } from "./constants";
+import { GeneratedDataCollectionClient } from "./generated";
+import { FailedLogsIngestionError, UploadOptions, UploadResult } from "./models";
+import { GZippingPolicy } from "./gZippingPolicy";
+import asyncPool from "tiny-async-pool";
+
+/**
+ * Options for Montior Logs Ingestion Client
+ */
+export interface LogsIngestionClientOptions extends CommonClientOptions {
+  /** Api Version */
+  apiVersion?: string;
+}
+const defaultIngestionScope = "https://monitor.azure.com//.default";
+
+/**
+ * Client for Monitor Logs Ingestion
+ */
+export class LogsIngestionClient {
+  /**
+   * Overrides client endpoint.
+   */
+  private endpoint: string;
+  private _dataClient: GeneratedDataCollectionClient;
+  /**
+   * Construct a MonitorIngestionClient that can be used to query logs using the Log Analytics Query language.
+   *
+   * @param tokenCredential - A token credential.
+   * @param options - Options for the MonitorIngestionClient.
+   */
+  constructor(
+    endpoint: string,
+    tokenCredential: TokenCredential,
+    options?: LogsIngestionClientOptions
+  ) {
+    const credentialOptions = {
+      credentialScopes: defaultIngestionScope,
+    };
+    this.endpoint = endpoint;
+    const packageDetails = `azsdk-js-monitor-ingestion/${SDK_VERSION}`;
+    const userAgentPrefix =
+      options?.userAgentOptions && options?.userAgentOptions.userAgentPrefix
+        ? `${options?.userAgentOptions.userAgentPrefix} ${packageDetails}`
+        : `${packageDetails}`;
+    this._dataClient = new GeneratedDataCollectionClient(tokenCredential, this.endpoint, {
+      ...options,
+      credentialScopes: credentialOptions?.credentialScopes ?? defaultIngestionScope,
+      userAgentOptions: {
+        userAgentPrefix,
+      },
+    });
+    // adding gziping policy bcz this is a single method client which needs gzipping
+    this._dataClient.pipeline.addPolicy(GZippingPolicy);
+  }
+
+  /**
+   * See error response code and error response message for more detail.
+   * @param ruleId - The immutable Id of the Data Collection Rule resource.
+   * @param streamName - The streamDeclaration name as defined in the Data Collection Rule.
+   * @param logs - An array of objects matching the schema defined by the provided stream.
+   * @param options - The options parameters.
+   */
+  async upload(
+    ruleId: string,
+    streamName: string,
+    logs: Record<string, any>[],
+    options?: UploadOptions
+  ): Promise<UploadResult> {
+    // TODO: Do we need to worry about memory issues when loading data for 100GB ?? JS max allocation is 1 or 2GB
```

Comment:
I think we should support an overload for streaming log objects (or strings), see how we do streaming upload in storage-blob.

Reply:
I'll note this down for beta-2

Reply:
https://github.com/Azure/azure-sdk-for-js/issues/22615 Filed